tests/arithm/tarithm.nim

```diff
@@ -135,9 +154,9 @@ block tshl:
     const t1: uint16 = 1'u16 shl 16
     const t2: uint32 = 1'u32 shl 32
     const t3: uint64 = 1'u64 shl 64
-    doAssert t0 == 0
-    doAssert t1 == 0
-    doAssert t2 == 0
+    doAssert t0 == 1
+    doAssert t1 == 1
+    doAssert t2 == 1
```

Comment:
We should be consistent with C as mentioned in https://github.com/nim-lang/Nim/issues/11138

For example uint32 numbers are implicitly modulo 2^32
`2^32 mod 2^32 == 0` not 1

This makes Nim VM unusable for big int libraries like [stint](https://github.com/status-im/nim-stint) or [constantine](https://github.com/status-im/nim-constantine)

Reply:
You misunderstand. The right operand gets the modulo operation. ``1'u32 shl 32`` = ``1'u32 shl (32 mod 32)`` = ``1'u32 shl 0`` = ``1'u32``. 

Here is a motivation for this change from [cppreference.com](https://en.cppreference.com/w/cpp/language/operator_arithmetic)

> In any case, if the value of the right operand is negative or is greater or equal to the number of bits in the promoted left operand, the behavior is undefined.

In other words ``1'u64 shl 64`` was mapped to undefined behavior in c/c++.